java/shopping-cart/src/main/java/org/apache/flink/statefun/examples/shoppingcart/UserShoppingCartFn.java

```diff
@@ -0,0 +1,164 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.flink.statefun.examples.shoppingcart;
+
+import static org.apache.flink.statefun.examples.shoppingcart.Identifiers.RECEIPT_EGRESS;
+import static org.apache.flink.statefun.examples.shoppingcart.Messages.*;
+import static org.apache.flink.statefun.examples.shoppingcart.Messages.ItemAvailability.*;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+import java.util.stream.Collectors;
+import org.apache.flink.statefun.sdk.java.AddressScopedStorage;
+import org.apache.flink.statefun.sdk.java.Context;
+import org.apache.flink.statefun.sdk.java.StatefulFunction;
+import org.apache.flink.statefun.sdk.java.TypeName;
+import org.apache.flink.statefun.sdk.java.ValueSpec;
+import org.apache.flink.statefun.sdk.java.io.KafkaEgressMessage;
+import org.apache.flink.statefun.sdk.java.message.Message;
+import org.apache.flink.statefun.sdk.java.message.MessageBuilder;
+import org.apache.flink.statefun.sdk.java.types.SimpleType;
+import org.apache.flink.statefun.sdk.java.types.Type;
+
+final class UserShoppingCartFn implements StatefulFunction {
+
+  static final TypeName TYPE = TypeName.typeNameFromString("com.example.fns/user-shopping-cart");
+
+  
+  static final ValueSpec<Basket> BASKET = ValueSpec.named("basket").withCustomType(Basket.TYPE);
+
+  @Override
+  public CompletableFuture<Void> apply(Context context, Message message) {
+    if (message.is(ADD_TO_CART)) {
+      AddToCart addToCart = message.as(ADD_TO_CART);
+
+      RequestItem requestItem = new RequestItem(addToCart.getQuantity());
+      Message request =
+          MessageBuilder.forAddress(InventoryFn.TYPE, Integer.toString(addToCart.getItemId()))
+              .withCustomType(REQUEST_ITEM_TYPE, requestItem)
+              .build();
+      context.send(request);
+    }
```

Comment:
What do you think about using if-else here, or `if (..) { ... return; }` ?

Reply:
Do you mean to add `return context.done();` to all of the individual `if` code blocks to make it explicitly clear that they are mutually exclusive?